src/main.ts

```diff
@@ -1,7 +1,7 @@
 import * as core from '@actions/core'
 import {getInputs} from './inputs-artifacts'
 
-async function run(): Promise<void> {
```

Comment:
Why ?

Reply:
For the exception raised, I have an idee.
It may not be catch if we don t wait the fonction to end.

And it raise in the void

Reply:
Same question: why "async" ? 

It is forbidden by node to call run() with await, so I figured out it was not necessary, and in the end, it makes no sense for your main() function that you call first before everything.